crates/compiler/load_internal/tests/test_load.rs

```diff
@@ -1563,6 +1599,325 @@ fn cannot_use_original_name_if_imported_with_alias() {
+                r#"
+        module [example]
+
+        import Api { key: 123 }
+
+        example = Api.url
+            "#
+            ),
+        ),
+    ];
+
+    let result = multiple_modules("module_params_typecheck_fail", modules).unwrap_err();
+    assert_eq!(
+        result,
+        indoc!(
+            r#"
+            ── IMPORT PARAMS MISMATCH in tmp/module_params_typecheck_fail/Main.roc ─────────
+
+            Something is off with the params provided by this import:
+
+            3│  import Api { key: 123 }
+                           ^^^^^^^^^^^^
+
+            This is the type I inferred:
+
+                { key : Num * }
+
+            However, Api expects:
+
+                { key : Str }
+            "#
+        )
+    );
+}
+
+#[test]
+fn module_params_missing_fields() {
+    let modules = vec![
+        (
+            "Api.roc",
+            indoc!(
+                r#"
+            module { key } -> [url]
+
+            url = "example.com/$(key)"
+            "#
+            ),
+        ),
+        (
+            "Main.roc",
+            indoc!(
+                r#"
+        module [example]
+
+        import Api {}
+
+        example = Api.url
+            "#
+            ),
+        ),
+    ];
+
+    let result = multiple_modules("module_params_missing_fields", modules).unwrap_err();
+    assert_eq!(
+        result,
+        indoc!(
+            r#"
+            ── IMPORT PARAMS MISMATCH in tmp/module_params_missing_fields/Main.roc ─────────
+
+            Something is off with the params provided by this import:
+
+            3│  import Api {}
+                           ^^
+
+            This is the type I inferred:
+
+                {}
+
+            However, Api expects:
+
+                { key : Str }
+
+            Tip: Looks like the key field is missing.
+            "#
+        )
+    );
+}
+
+#[test]
+fn module_params_extra_fields() {
+    let modules = vec![
+        (
+            "Api.roc",
+            indoc!(
+                r#"
+            module { key } -> [url]
+
+            url = "example.com/$(key)"
+            "#
+            ),
+        ),
+        (
+            "Main.roc",
+            indoc!(
+                r#"
+        module [example]
+
+        import Api { key: "123", doesNotExist: Bool.true }
+
+        example = Api.url
+            "#
+            ),
+        ),
+    ];
+
+    let result = multiple_modules("module_params_extra_fields", modules).unwrap_err();
+    assert_eq!(
+        result,
+        indoc!(
+            r#"
+            ── IMPORT PARAMS MISMATCH in tmp/module_params_extra_fields/Main.roc ───────────
+
+            Something is off with the params provided by this import:
+
+            3│  import Api { key: "123", doesNotExist: Bool.true }
+                           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+            This is the type I inferred:
+
+                { doesNotExist : Bool, … }
+
+            However, Api expects:
+
+                { … }
```

Comment:
This error message isn't great, but it seems to be a general issue with extra fields in records. I'll investigate separately.